components/eamxx/src/physics/share/physics_constants.hpp

```diff
@@ -126,6 +126,29 @@ struct Constants
   static constexpr Scalar earth_ellipsoid3 = 1.175;     // third expansion coefficient for WGS84 ellipsoid
 };
 
+template <typename Scalar>
+struct P3_Constants
+{
+
+private:
+   static Scalar p3_autoconversion_factor;
```

Comment:
One thing to watch out for is static variables (I don't mean `constexpr`s, but actual variables) that are accessed on both host and GPU. Without Cuda/HIP-specific annotations, that won't work, and it's a little messy to do that. The simplest portable way to handle a struct of set-able parameters is to make them non-static. You can pass a struct of non-static variables around and to device code like any other variable.